imports/plugins/included/product-variant/containers/productsContainer.js

```diff
@@ -1,217 +1,33 @@
-      event.preventDefault();
-      this.setState({
-        initialLoad: false
-      });
-      loadMoreProducts();
-    }
-
-    render() {
-      return (
-        <Comp
-          {...this.props}
-          ready={this.ready}
-          loadMoreProducts={this.loadMoreProducts}
-          loadProducts={this.loadProducts}
-        />
-      );
-    }
-  }
-);
+ProductsContainer.propTypes = {
+  isAdmin: PropTypes.bool
+};
 
 function composer(props, onData) {
-  window.prerenderReady = false;
-
-  let canLoadMoreProducts = false;
-
-  const slug = Reaction.Router.getParam("slug");
-  const shopIdOrSlug = Reaction.Router.getParam("shopSlug");
-
-  const tag = Tags.findOne({ slug }) || Tags.findOne(slug);
-  const scrollLimit = Session.get("productScrollLimit");
-  let tags = {}; // this could be shop default implementation needed
-  let shopIds = {};
-
-  if (tag) {
-    tags = { tags: [tag._id] };
-  }
-
-  if (shopIdOrSlug) {
-    shopIds = { shops: [shopIdOrSlug] };
-  }
-
-  // if we get an invalid slug, don't return all products
-  if (!tag && slug) {
-    onData(null, {
-      showNotFound: true
-    });
-
-    return;
-  }
-
-  const currentTag = ReactionProduct.getTag();
-
-  const sort = {
-    [`positions.${currentTag}.position`]: 1,
-    [`positions.${currentTag}.createdAt`]: 1,
-    createdAt: 1
-  };
-
-  const viewAsPref = Reaction.getUserPreferences("reaction-dashboard", "viewAs");
-
-  // Edit mode is true by default
-  let editMode = true;
-
-  // if we have a "viewAs" preference and the preference is not set to "administrator", then edit mode is false
-  if (viewAsPref && viewAsPref !== "administrator") {
-    editMode = false;
-  }
-
-  const queryParams = Object.assign({}, tags, Reaction.Router.current().query, shopIds);
-  const productsSubscription = Meteor.subscribe("Products", scrollLimit, queryParams, sort, editMode);
-
-  if (productsSubscription.ready()) {
-    window.prerenderReady = true;
-  }
-
-  const activeShopsIds = Shops.find({
-    $or: [
-      { "workflow.status": "active" },
-      { _id: Reaction.getPrimaryShopId() }
-    ]
-  }).map((activeShop) => activeShop._id);
-
-  const productCursor = Products.find({
-    ancestors: [],
-    type: { $in: ["simple"] },
-    shopId: { $in: activeShopsIds }
-  });
-
-  const sortedProducts = ReactionProduct.sortProducts(productCursor.fetch(), currentTag);
-  Session.set("productGrid/products", sortedProducts);
-
-  const productIds = [];
-  const stateProducts = sortedProducts.map((product) => {
-    productIds.push(product._id);
-
-    return {
-      ...applyProductRevision(product)
-    };
-  });
-
-  reactiveProductIds.set(productIds);
-
-  canLoadMoreProducts = productCursor.count() >= Session.get("productScrollLimit");
-
-  const isActionViewOpen = Reaction.isActionViewOpen();
-  if (isActionViewOpen === false) {
-    Session.set("productGrid/selectedProducts", []);
-  }
+  const isAdmin = Reaction.hasPermission("createProduct", Meteor.userId());
```

Comment:
Can we do this outside of a composer, and then get rid of the composer since it's the only thing inside of it? Or does the admin check need to be run inside a composer?

Reply:
It's a reactive data source, so if your permissions change, the grid can reflect that change with the admin/customer product grid.